gluten-core/src/main/scala/io/glutenproject/expression/TernaryExpressionTransformer.scala

```diff
@@ -0,0 +1,243 @@
+      throw new UnsupportedOperationException(s"Not supported yet.")
+    }
+
+    val functionMap = args.asInstanceOf[java.util.HashMap[String, java.lang.Long]]
+    val functionName = ConverterUtils.makeFuncName(ConverterUtils.REPLACE,
+      Seq(str.dataType, search.dataType, replace.dataType), FunctionConfig.OPT)
+    val functionId = ExpressionBuilder.newScalarFunction(functionMap, functionName)
+    val expressionNodes = Lists.newArrayList(strNode, searchNode, replaceNode)
+    val typeNode = TypeBuilder.makeString(original.nullable)
+    ExpressionBuilder.makeScalarFunction(functionId, expressionNodes, typeNode)
+  }
+}
+
+class SplitTransformer(str: Expression, delimiter: Expression, limit: Expression,
+                       original: Expression)
+  extends StringSplit(str: Expression, delimiter: Expression, limit: Expression)
+    with ExpressionTransformer
+    with Logging {
+
+  override def doTransform(args: java.lang.Object): ExpressionNode = {
+    throw new UnsupportedOperationException("Not supported: Split.")
+  }
+}
+
+class SubStringTransformer(str: Expression, pos: Expression, len: Expression, original: Expression)
+  extends Substring(str: Expression, pos: Expression, len: Expression)
+    with ExpressionTransformer
+    with Logging {
+
+  override def doTransform(args: java.lang.Object): ExpressionNode = {
+    val strNode =
+      str.asInstanceOf[ExpressionTransformer].doTransform(args)
+    val posNode =
+      pos.asInstanceOf[ExpressionTransformer].doTransform(args)
+    val lenNode =
+      len.asInstanceOf[ExpressionTransformer].doTransform(args)
+
+    if (!strNode.isInstanceOf[ExpressionNode] ||
+      !posNode.isInstanceOf[ExpressionNode] ||
+      !lenNode.isInstanceOf[ExpressionNode]) {
+      throw new UnsupportedOperationException(s"not supported yet.")
+    }
+
+    val functionMap = args.asInstanceOf[java.util.HashMap[String, java.lang.Long]]
+    val functionName = ConverterUtils.makeFuncName(
+      ConverterUtils.SUBSTRING, Seq(str.dataType), FunctionConfig.OPT)
+    val functionId = ExpressionBuilder.newScalarFunction(functionMap, functionName)
+    val expressionNodes = Lists.newArrayList(
+      strNode.asInstanceOf[ExpressionNode],
+      posNode.asInstanceOf[ExpressionNode],
+      lenNode.asInstanceOf[ExpressionNode])
+    // Substring inherits NullIntolerant, the output is nullable
+    val typeNode = TypeBuilder.makeString(true)
+
+    ExpressionBuilder.makeScalarFunction(functionId, expressionNodes, typeNode)
+  }
+}
+
+object TernaryExpressionTransformer {
+
+  def create(first: Expression, second: Expression, third: Expression,
+    original: Expression): Expression = original match {
+      case _: StringLocate =>
+        // locate() gets incorrect results, so fall back to Vanilla Spark
+        throw new UnsupportedOperationException("Not supported: locate().")
+      case _: StringSplit =>
+        // split() gets incorrect results, so fall back to Vanilla Spark
```

Comment:
It can be fallbacked to vanilla with some changes in the native validator, but the conversion to Substrait plan could be supported.

Reply:
Yes, I will open another PR to support this.